### PR TITLE
filamat and matinfo: add more diagnostics.

### PR DESCRIPTION
**Pull request by prideout**
These features help with diagnosing SPIR-V related issues.

### CODE REVIEW

File: tools/matinfo/src/main.cpp [hunk 1]
```diff
@@ -866,15 +878,169 @@ static void transpileSpirv(const std::vector<uint32_t>& spirv) {
...
+}
+
+static void analyzeSpirv(const std::vector<uint32_t>& spirv, const char* disassembly) {
+    using namespace std;
+
+    const map<int, string> glsl = transpileSpirvToLines(spirv);
+    const regex globalDecoratorPattern("OpDecorate (\\%[A-Za-z_0-9]+) RelaxedPrecision");
+    const regex typeDefinitionPattern("(\\%[A-Za-z_0-9]+) = OpType[A-Z][a-z]+");
+    const regex memberDecoratorPattern("OpMemberDecorate (\\%[A-Za-z_0-9]+) ([0-9]+) RelaxedPrecision");
+    const regex structDefinitionPattern("(\\%[A-Za-z_0-9]+) = OpTypeStruct");
```
Comment (bejado) [inline tools/matinfo/src/main.cpp]: I don't see where you're using `structDefinitionPattern` anywhere?

File: libs/filamat/src/GLSLPostProcessor.cpp [hunk 2]
```diff
@@ -218,7 +218,12 @@ bool GLSLPostProcessor::process(const std::string& inputShader,
     switch (mOptimization) {
         case MaterialBuilder::Optimization::NONE:
             if (mSpirvOutput) {
-                GlslangToSpv(*program.getIntermediate(mShLang), *mSpirvOutput);
+                SpvOptions options;
+#ifndef NDEBUG
+                options.generateDebugInfo = true;
```
Comment (bejado) [inline libs/filamat/src/GLSLPostProcessor.cpp]: Are these options necessary for the `analyze-spirv` flag? If so, maybe we should only allow `analyze-spirv` in debug builds of `matinfo`.

File: tools/matinfo/src/main.cpp [hunk 3]
```diff
@@ -132,24 +137,26 @@ static void printUsage(const char* name) {
...
-                    "       Print the nth Vulkan shader transpiled into GLSL\n\n"
-                    "   --dump-binary=[index], -b\n"
-                    "       Dump binary SPIRV for the nth Vulkan shader to 'out.spv'\n\n"
-                    "   --license\n"
-                    "       Print copyright and license information\n\n"
+            "Usage:\n"
+            "    MATINFO [options] <material file>\n"
+            "\n"
+            "Options:\n"
+            "   --analyze-spirv=[index], -a\n"
```
Comment (romainguy) [inline tools/matinfo/src/main.cpp]: Can you move this option at the end of the help message? It's really for us and shouldn't come first (especially not before the more common options)

File: tools/matinfo/src/main.cpp (hunk 1 above)
Comment (prideout) [reply tools/matinfo/src/main.cpp]: Removed.

File: libs/filamat/src/GLSLPostProcessor.cpp (hunk 2 above)
Comment (prideout) [reply libs/filamat/src/GLSLPostProcessor.cpp]: This is orthogonal to the matinfo changes.

File: tools/matinfo/src/main.cpp (hunk 3 above)
Comment (prideout) [reply tools/matinfo/src/main.cpp]: Fixed.

**Pull request merged**